app/src/main/java/com/hedvig/app/authenticate/LoginStatusService.kt

```diff
@@ -47,9 +46,14 @@ class SharedPreferencesLoginStatusService(
         yield()
         return when {
             isLoggedIn -> LoginStatus.LoggedIn
-            isViewingOffer -> LoginStatus.InOffer(
-                quoteCartId = sharedPreferences.getString("quoteCartId", null)?.let { QuoteCartId(it) }
-            )
+            isViewingOffer -> {
+                val quoteCartId = sharedPreferences.getString("quoteCartId", null)?.let(::QuoteCartId)
+                if (quoteCartId != null) {
+                    LoginStatus.InOffer(quoteCartId)
+                } else {
+                    LoginStatus.Onboarding
+                }
```

Comment:
This is the most meaningful change, since in the SplashScreen, if this returns Onboarding it won't try to launch the Offer screen with a null QuoteCartId